.github/workflows/build_and_release.yml

```diff
@@ -0,0 +1,59 @@
+name: Build and Release
+
+on:
+  workflow_dispatch: # Allows manual triggering
+    inputs:
+      tag:
+        description: 'Tag name'     
+        required: true
+      note:
+        description: 'Release note'
+        
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - name: Build
+        run: |
+          npm i -g @vercel/ncc
+          npm i @actions/github
+          npm run build
+      - name: Commit
+        run: |
+          git config --global user.name 'yulin17-chang'
+          git config --global user.email '182771494+yulin17-chang@users.noreply.github.com'
+          git commit -am "dist build"
+          git push
+
+  release:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v4
+      - name: Pull latest
+        run: |
+          git config --global user.name 'yulin17-chang'
+          git config --global user.email '182771494+yulin17-chang@users.noreply.github.com'
```

Comment:
See my above comment for setting the Git user name and email dynamically using the workflow’s executor name and the no-reply email format.